src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/ChangeManager.java

```diff
@@ -52,21 +56,28 @@ private <P, C, S, U> boolean compareAndBuildChange(
       BiPredicate<P, C> equalityPredicate,
       BiConsumer<T, S> dataSetter) {
 
-    var convertedProposedData = proposedConverter.apply(proposedData);
-
-    if (leftOrRightDataFromPair == null && objectToAddChanges == null) {
-      return false;
+    S convertedProposedData = null;
+    if (proposedConverter != null && proposedData != null) {
```

Comment:
I feel the below is becoming way too complex, is there a way to do the null checks and return false up front?

Example:

```
   if (proposedConverter == null || (proposedData == null || isEmpty(proposedData))) {
      return false;
    }
```

Reply:
Yeah you're right 

Though `proposedData == null` isn't needed as it's tested in isEmpty()

Reply:
That was just an example, there may be other checks to add to this. The same for the block where we throw an exception